qiskit/utils/multiprocessing.py

```diff
@@ -27,13 +26,18 @@ def local_hardware_info():
     Returns:
         dict: The hardware information.
     """
+
+    if hasattr(os, "sched_getaffinity"):
+        num_cpus = int(len(os.sched_getaffinity(0)) / 2) or 1
+    else:
+        num_cpus = 1
```

Comment:
I think in this case we can do:

```suggestion
        num_cpus = os.cpu_count()
        if num_cpus is None:
            num_cpus = 1
        else:
            num_cpus = int(num_cpus / 2) or 1
```